kong/db/dao/snis.lua

```diff
@@ -18,6 +18,31 @@ local function invalidate_cache(self, old_entity, err, err_t)
 end
 
 
+local function check_private_key_exists(self, sni)
+  if not sni.certificate or not sni.certificate.id then
+    -- just let validator do its job
+    return true
+  end
+
+  local cert, err, err_t = self.db.certificates:select({
+    id = sni.certificate.id,
+  })
+  if not cert then
+    return nil, err, err_t
+  end
+
+  if not cert.key then
+    local err_t = self.errors:schema_violation({
+      certificate = "must have the \"key\" field (private key) set to" ..
```

Comment:
We should also fix this message (i.e. quotes and wording)

Reply:
Changed to `"certificate: cannot be used by SNI because specified Certificate does not have the 'key' attribute set"`.